letslambda.py

```diff
@@ -142,127 +348,217 @@ def answer_dns_challenge(client, domain, challenge):
+
+    try:
+        (certificate, ar) = client.poll_and_request_issuance(ComparableX509(csr), [auth_resource])
+    except errors.PollError as e:
+        LOG.error("Failed to get certificate issuance for '{0}'.".format(domain['name']))
+        LOG.error("Error: {0}".format(e))
+        print e
+        return (False, False, False)
 
-def createIAMCertificate(domain, certificate, key):
-    """
-    Create a new IAM certificate from ACME and private key.
-    It also fetched the chain certificate from ACME if provided
-    """
-    # First we fetch the chain certificate
     chain = requests.get(certificate.cert_chain_uri)
     chain_certificate = None
+
     if chain.status_code == 200:
         chain_certificate = crypto.load_certificate(crypto.FILETYPE_ASN1, chain.content)
+        pem_chain_certificate = crypto.dump_certificate(crypto.FILETYPE_PEM, chain_certificate).decode("ascii")
+    else:
+        LOG.error("Failed to retrieve chain certificate. Status was '{0}'.".format(chain.status_code))
+        pem_chain_certificate = False
 
-    iam_connection = iam.connect_to_region(exec_region)
-    res = iam_connection.upload_server_cert(
-        domain['name'] + "-" + datetime.utcnow().strftime("%Y-%m-%dT%H-%M"),
-        crypto.dump_certificate(crypto.FILETYPE_PEM, certificate.body.wrapped).decode("ascii"),
-        crypto.dump_privatekey(crypto.FILETYPE_PEM, key).decode("ascii"),
-        crypto.dump_certificate(crypto.FILETYPE_PEM, chain_certificate).decode("ascii"),
-        "/")
-    sleep(10)
+    pem_certificate = crypto.dump_certificate(crypto.FILETYPE_PEM, certificate.body.wrapped).decode("ascii")
+    pem_private_key = crypto.dump_privatekey(crypto.FILETYPE_PEM, key).decode("ascii")
 
-    return res
-
-def updateELB(conf, iam_certificate):
-    LOG.info("Updating ELB with new certificate")
-    elb_connection = elb.connect_to_region(conf['elb_region'])
-    response = elb_connection.set_lb_listener_SSL_certificate(conf['elb'], 443, iam_certificate['upload_server_certificate_response']['upload_server_certificate_result']['server_certificate_metadata']['arn'])
-    return response
+    return (pem_chain_certificate, pem_certificate, pem_private_key)
 
 def lambda_handler(event, context):
-    bucket = event['bucket']
-    region = event['region']
+    if 'bucket' not in event:
+        LOG.error("No bucket name has been provided. Exiting.")
+        exit(1)
+    s3_bucket = event['bucket']
 
-    LOG.info("Retrieving configuration file from bucket '{0}' in region '{1}' ".format(bucket, region))
-    connection = s3.connect_to_region(region, calling_format=OrdinaryCallingFormat())
-    try:
-        bucket = connection.get_bucket(bucket);
-    except S3ResponseError as e:
-        print(e)
-        LOG.error("Cannot fetch bucket : {}".format(bucket))
+    if 'region' not in event:
+        LOG.error("No bucket region has been provided. Exiting.")
         exit(1)
+    s3_region = event['region']
```

Comment:
If the region is not specified, I would expect to take the local one. Less config effort.